templates/terraform/resource.erb

```diff
@@ -204,13 +204,26 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
     d.SetId(id)
 
 <%  if !object.async.nil? && object.async.allow?('create') -%>
+<%  if object.async.is_a? Api::PollAsync -%>
```

Comment:
Instead of glomping these together under a single check on async, if these are both top-level we can set them up as two completely separate hunks of code.

We have:

```
if !object.async.nil? && object.async.allow?('create')
  if object is OpAsync
    ... do op stuff ...
  else
    ... do poll stuff ...
  end

  ... do shared stuff ...
  if !object is PollAsync
    ... do op stuff ...
  end
  ... do shared stuff ...
end
```

But given that change it would feel really natural to do:

```
if !object.op_async.nil? && object.op_async.allow?('create')
 ... do op stuff ...
end

if !object.poll_async.nil? && object.poll_async.allow?('create')
 ... do poll stuff ...
end
```

Not only that, but if/when we support resources with LROs and resource state, they should just work.